index.js

```diff
@@ -5,8 +5,31 @@
  * MIT Licensed
  */
 var promiseExists = typeof Promise === 'function';
-var globalObject = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : self; // eslint-disable-line
-var isDom = 'location' in globalObject && 'document' in globalObject;
+
+/* eslint-disable */
+// See http://stackoverflow.com/a/6930376
+var globalObject;
+try {
+  globalObject = Function('return this')() || (42, eval)('this');
+} catch (e) {
+  globalObject = window;
+}
+/* eslint-enable */
+
+/*
+ * All of these attributes must be available on the global object for the current environment
+ * to be considered a DOM environment (browser)
+ */
+var domIndicatorAttributes = [
+  'location',
+  'document',
+  'navigator',
+  'HTMLElement',
+];
```

Comment:
One thing I'm concerned about here is that lines like this `if (obj === (globalObject.navigator || {}).mimeTypes) {` indicate that sometimes `globalObject.navigator` isn't expected to be defined, even in a valid browser environment. However, this isn't my area of expertise. @keithamus @shvaikalesh thoughts?